python/rgbmatrix/core.pyx

```diff
@@ -83,18 +83,100 @@ cdef class FrameCanvas(Canvas):
         def __get__(self): return (<cppinc.FrameCanvas*>self.__getCanvas()).pwmbits()
         def __set__(self, pwmBits): (<cppinc.FrameCanvas*>self.__getCanvas()).SetPWMBits(pwmBits)
 
+cdef class RGBMatrixOptions:
+    def __cinit__(self):
```

Comment:
I do think it would be very neat if the RGBMatrixOptions have its own argument parsing for things it is interested in. These are very specific as `--led-rows` or `--led-pwm-lsb-nanoseconds`. Of course, it only works well if Python has a standard support to do these kind of things; it sounds like with the argument parsing thing build into Python it is possible to delegate part of it to other subsystems (but then again, I am not a Python expert. In C, it is a bit awkard, but it commonly done this way for complicated subsystems that pick out arguments).

So the **library** should provide and deal with command line options it is responsible for. That way, the application also does not have to deal with new options that are added over time. The flag namespace starting with `--led-` makes it not clash.

The **application** should deal with options that are relevant for its application doing, such as `--imagepath`.

If the application is interested in providing all the options by themself and not call the RGBMatrixOption matrix parsing code, they can do that as well. But from a using-an-API point of view it is nicer if there is a default way to do this, without having to add the same options every single time.

Look at the current example: they are all deriving from samplebase, mostly because they all have to deal with parsing the options. If they could just call `options.InitFromFlags(self.parser)`, this would be much easier to deal with for someone who wants to quickly see results.